language/testing-infra/module-generation/src/generator.rs

```diff
@@ -160,27 +165,39 @@ impl<'a> ModuleGenerator<'a> {
         }
     }
 
-    fn type_parameters(&mut self) -> Vec<(TypeVar, BTreeSet<Ability>)> {
+    fn fun_type_parameters(&mut self) -> Vec<(TypeVar, BTreeSet<Ability>)> {
         // Don't generate type parameters if we're generating simple types only
         if self.options.simple_types_only {
             vec![]
         } else {
             let num_ty_params = self.index(self.options.max_ty_params);
             let abilities = BTreeSet::from_iter(vec![Ability::Copy, Ability::Drop]);
-            init!(
-                num_ty_params,
-                (
-                    Spanned::unsafe_no_loc(TypeVar_::new(self.identifier())),
-                    abilities.clone(),
-                )
-            )
+            init!(num_ty_params, {
+                let name = Spanned::unsafe_no_loc(TypeVar_::new(self.identifier()));
```

Comment:
bit weird to nest the name like this unless you have to. Might look better as 
```
let name = Spanned....
init!(num_ty_params, (name, abilities.clone()))
```

Reply:
you have to because `self.identifier()` has side effects and you want to call it in each iteration.

Reply:
oof. Alright I guess keep it as is then